freight/vcsremote/github.py

```diff
@@ -14,11 +100,99 @@ def repo_url(self) -> str:
 
         if result is None:
             raise RuntimeError(
-                "GitHub repo URL does not match expected format giithub.com:owner/repo.git"
+                "GitHub repo URL does not match expected format github.com:owner/repo.git"
             )
 
         repo = result.group("repo")
         return f"https://github.com/{repo}"
 
+    @property
+    def repo_parts(self) -> Tuple[str, str]:
+        result = re.search(r"(?<=:)((?P<owner>[^\/]+)\/(?P<repo>[^.]+))", self.repo.url)
+
+        if result is None:
+            raise RuntimeError(
+                "GitHub repo URL does not match expected format github.com:owner/repo.git"
+            )
+
+        return (result.group("owner"), result.group("repo"))
+
     def get_commit_url(self, sha: str) -> str:
         return f"{self.repo_url}/commit/{sha}"
+
+    def get_commits_info(self, shas: List[str]):
+        """
+        Retrieves a list of changes, including pull request information, for a
+        range of commit shas.
+
+        The shas should be a contiguous list of shas from the git
+        repository.
+        """
+        token = current_app.config["GITHUB_TOKEN"]
+        headers = {
+            "Authorization": f"token {token}",
+        }
+
+        if len(shas) == 0:
+            return []
+
+        repo_owner, repo_name = self.repo_parts
+        query = build_commit_query(repo_owner, repo_name, shas)
+
+        commits_resp = http.post(
+            "https://api.github.com/graphql",
+            json=query,
+            headers=headers,
+        )
+        commits_dict = {
```

Comment:
There's not a lot of error handling here. How would Freight deal with an error being thrown?

Reply:
Yeah definitely we don't have a lot of error handling for this right now.

The API will return a 500 if it blows up, and the error will end up in freight.

This shouldn't affect the frontend adversely, it will just show an error.